### PR TITLE
Moving appconfig appflow appintegrations

### PR DESCRIPTION
**Pull request by svscheg**
### Description of your changes
Moving 6 resources in the `appconfig` group:

- [`aws_appconfig_application`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_application)
- [`aws_appconfig_configuration_profile`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_configuration_profile)
- [`aws_appconfig_deployment`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_deployment)
- [`aws_appconfig_deployment_strategy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_deployment_strategy)
- [`aws_appconfig_environment`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_environment)
- [`aws_appconfig_hosted_configuration_version`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_hosted_configuration_version)

Moving 1 resources in the `appflow` group:

- [`aws_appflow_flow`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appflow_flow)

Moving 1 resource in the `appintegrations` group:

- [`aws_appintegrations_event_integration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appintegrations_event_integration)

Fixes https://github.com/upbound/provider-aws/issues/23

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manually
- [`aws_appconfig_deployment`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_deployment)
Screenshot 2022-11-30 at 12 34 39

Uptest
Moving 6 resources in the `appconfig` group:

- [`aws_appconfig_application`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_application) - https://github.com/upbound/provider-aws/actions/runs/3572128766/jobs/6004666277
- [`aws_appconfig_configuration_profile`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_configuration_profile) - https://github.com/upbound/provider-aws/actions/runs/3572129897/jobs/6004668855
- [`aws_appconfig_deployment_strategy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_deployment_strategy) - https://github.com/upbound/provider-aws/actions/runs/3572131293/jobs/6004671614
- [`aws_appconfig_environment`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_environment) - https://github.com/upbound/provider-aws/actions/runs/3572268059/jobs/6004964311
- [`aws_appconfig_hosted_configuration_version`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_hosted_configuration_version) - https://github.com/upbound/provider-aws/actions/runs/3572143803/jobs/6004699243

Moving 1 resources in the `appflow` group:

- [`aws_appflow_flow`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appflow_flow) - https://github.com/upbound/provider-aws/actions/runs/3631956827/jobs/6127235702

Moving 1 resource in the `appintegrations` group:

- [`aws_appintegrations_event_integration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appintegrations_event_integration) - https://github.com/upbound/provider-aws/actions/runs/3631963079/jobs/6127250242

### CODE REVIEW

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/deployment.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appflow/flow.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/application.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/configurationprofile.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/deploymentstrategy.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/environment.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/hostedconfigurationversion.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appintegrations/eventintegration.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/environment.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appflow/flow.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/appconfig/deployment.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -1271,6 +1271,31 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
+
+	// appintegrations
+	//
+	// Amazon AppIntegrations Event Integrations can be imported using the name
+	"aws_appintegrations_event_integration": config.NameAsIdentifier,
+
+	// appflow
+	//
+	// AppFlow flows can be imported using the arn
+	"aws_appflow_flow": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalname.go:1298]: As above:
```suggestion
	// arn:aws:appflow:us-west-2:123456789012:flow/example-flow
	"aws_appflow_flow": config.TemplatedStringAsIdentifier("name", "arn:aws:appflow:{{ .parameters.region }}:{{ .client_metadata.account_id }}:flow/{{ .external_name }}"),
```

File: config/externalnamenottested.go [hunk 2]
```diff
@@ -25,30 +25,10 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
 	//
-	// AppFlow flows can be imported using the arn
-	"aws_appflow_flow": config.IdentifierFromProvider,
-
-	// appintegrations
-	//
-	// Amazon AppIntegrations Event Integrations can be imported using the name
-	"aws_appintegrations_event_integration": config.NameAsIdentifier,
+	// AppFlow connector profiles can be imported using the arn
+	"aws_appflow_connector_profile": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:31]: Why do we introduce a new untested configuration in this PR? This does not exist in `config/externalname.go`.

File: config/provider.go [hunk 3]
```diff
@@ -111,7 +111,6 @@ var skipList = []string{
 	"aws_alb_target_group$",            // identical with aws_lb_target_group.
 	"aws_alb_target_group_attachment$", // identical with aws_lb_target_group_attachment.
 	"aws_iot_authorizer$",              // failure with unknown reason.
-	"aws_appflow_connector_profile$",   // failure with unknown reason.
```
Comment (ulucinar) [inline config/provider.go:114]: I think we still don't generate this.

**Review by ulucinar — COMMENTED** [review]
Hi @svscheg,
Thank you for the PR. Left some comments to further normalize the API.

File: config/provider.go (hunk 3 above)
Comment (svscheg) [reply config/provider.go:114]: back "aws_appflow_connector_profile$",   // failure with unknown reason. to config/provider.go

File: config/externalnamenottested.go (hunk 2 above)
Comment (svscheg) [reply config/externalnamenottested.go:31]: deleted

**Comment by ulucinar** [comment]
/test-examples="examples/appflow/flow.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/appintegrations/eventintegration.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/appflow/flow.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/appintegrations/eventintegration.yaml"

**Review by ezgidemirel — APPROVED** [review]
Thanks a lot @svscheg and @ulucinar !!

**Pull request merged**